src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/StringReaderTest.kt

```diff
@@ -0,0 +1,156 @@
+package com.github.doyaaaaaken.kotlincsv.client
+
+import com.github.doyaaaaaken.kotlincsv.dsl.context.CsvReaderContext
+import com.github.doyaaaaaken.kotlincsv.dsl.csvReader
+import com.github.doyaaaaaken.kotlincsv.util.CSVFieldNumDifferentException
+import com.github.doyaaaaaken.kotlincsv.util.CSVParseFormatException
+import com.github.doyaaaaaken.kotlincsv.util.Const
+import com.github.doyaaaaaken.kotlincsv.util.MalformedCSVException
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.collect
+
+class StringReaderTest : WordSpec({
```

Comment:
The logic of this test class is duplicated with `CsvReaderTest`.
This class doesn't seem necessary because the test for `StringReaderImpl` is executed on `CsvReaderTest`.

Reply:
CsvReaderTest, as a JvmTest, is using the Java CsvReader that utilizes ByteInputStream, even on a String input (for example, CsvReader.kt line 26). So I don't think it is testing StringReaderImpl. I agree there is a lot of duplication. A more direct approach would be to test StringReaderImpl as a generic reader implementation and not necessarily tied to CSV reading - but duplicating much of CsvReader was a faster way to get test coverage and ensure it met all the requirements needed for CSV parsing.